osu.Game/Screens/OnlinePlay/Lounge/LoungeSubScreen.cs

```diff
@@ -310,6 +314,41 @@ private void onLeaving()
             });
         });
 
+        /// <summary>
+        /// Copies a room and opens it as a fresh (not-yet-created) one.
+        /// </summary>
+        /// <param name="room">The room to copy.</param>
+        public void OpenCopy(Room room)
+        {
+            Debug.Assert(room.RoomID.Value != null);
+
+            if (joiningRoomOperation != null)
+                return;
+
+            joiningRoomOperation = ongoingOperationTracker?.BeginOperation();
+
+            var req = new GetRoomRequest(room.RoomID.Value.Value);
+
+            req.Success += r =>
+            {
+                // ID must be unset as we use this as a marker for whether this is a client-side (not-yet-created) room or not.
+                r.RoomID.Value = null;
+
+                Open(r);
+
+                joiningRoomOperation?.Dispose();
+                joiningRoomOperation = null;
+            };
+
+            req.Failure += _ =>
+            {
+                joiningRoomOperation?.Dispose();
```

Comment:
```suggestion
            req.Failure += exception =>
            {
                Logger.Error(exception, "Couldn't create a copy of this playlist.");
                joiningRoomOperation?.Dispose();
```

Seems we should probably expose this error?